### ISSUE TITLE
Example config doesn't show how to use alternative port

### ISSUE THREAD

**Issue by aabdnn** [getdnsapi/stubby#15]
The example stubby config shows how to list addresses to listen on, and says that the default port is 53, but doesn't explicitly show how to specify an alternative port using the address@port syntax. I eventually figured it out, but it would be nice if this were made clear in the example, because it's not obvious.

**Comment by wtoorop** [comment]
Yes, the @ works for both IPv4 and IPv6, with IPv4 you can also use a colon. I'd like to have that for IPv6 as well (i.e. [0::1]:8853 ), but having difficulties with the parser interpreting square brackets as list delimiters.  Finally it is also possible to use a dictionary for an address, with `address_data` and `port` keys, for example:

```sh
listen_addresses:
- 127.0.0.1:8853
- address_data: 0::1
  port: 8853
```

**Comment by aabdnn** [comment]
Multiple ways of representing listen addresses might be useful, but you're just adding even more code to handle the various formats. Why not just keep it simple, and decree that the listen addresses must be only a list, and must be specified as address or address@port? This makes it simple, and less confusing, and easier for you to maintain, because you don't have to deal with the various ways that someone might configure something.

**Comment by wtoorop** [comment]
That would be nice.  Unfortunately the getdns API spec defines the quite verbose dictionary format as the default.  So the original format without shortcuts would have been:

```sh
listen_address:
  - address_data: 127.0.0.1
  - address_type: "IPv4"
  - port: 8853
```

Note, that even in this example there are already some non-spec "improvements" as originally there was not a primitive presentation type for an IP address (neither for strings, domain name or Base64 data for that matter).

**Comment by aabdnn** [comment]
Hmm, the "address_type" key looks redundant to me. What happens when someone sets "address_data" to "127.0.0.1" and address_type to "IPv6"? :) The address_type should always be inferred from the address. I can't think of why anyone would need to specify the type.

Anyway, I understand your concern. You have to read the addresses, and give them to getdns in a dictionary. In that case, why not drop the shortcut format of "address@port" and just make "listen_addresses" a dictionary? Then there's only one way to configure it, and it's compatible with what getdns wants. Stubby is a getdns client, so configuring it as closely as possible to getdns is a virtue.

**Comment by saradickinson** [comment]
https://github.com/getdnsapi/stubby/pull/17 adds a comment describing how to use a @ for this

**Comment by saradickinson** [comment]
Fixed in commit a3529fb5541124f6fa29019b2df6f3933252338c
